lab/webapp/src/components/Dataset/components/DatasetMenu/components/Charts/BoxPlot/index.jsx

```diff
@@ -0,0 +1,199 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Header } from 'semantic-ui-react';
+import * as d3 from "d3";
+//import * as bananaSet from './bananaJson.json';
+
+class BoxPlot extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+
+    };
+  this.createBoxPlot = this.createBoxPlot.bind(this);
+  }
+
+  componentDidMount() {
+    this.createBoxPlot();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    //this.createBarGraph();
+  }
+
+  // from - https://www.d3-graph-gallery.com/graph/boxplot_basic.html
+  createBoxPlot(){
+    const { dataPreview, valByRowObj, rawKey, cleanKey } = this.props;
+    //window.console.log('datapreview for boxplot', dataPreview);
+    let margin = { top: 5, right: 60, bottom: 50, left: 85 },
+        width = 425 - margin.left - margin.right,
+        height = 300 - margin.top - margin.bottom;
+
+    // to make background of svg transparent set stroke & fill to none
+    // or do not specify background color
+    let svg = d3.select("#test_box_plot_" + cleanKey)
+    .append("svg")
+      .attr("width", width + margin.left + margin.right)
+      .attr("height", height + margin.top + margin.bottom)
+      .attr("stroke", "none")
+    .style("fill", "none")
+    .append("g")
+      .attr("transform",
+            "translate(" + margin.left + "," + margin.top + ")");
+    // get stats
+    let data_sorted = valByRowObj[rawKey].sort(d3.ascending);
+    // let data_sorted_AB = valByRowObj[rawKey].sort( (a, b) => {
+    //   return a - b;
+    // });
+
+    let q1 = d3.quantile(data_sorted, .25);
+    //let median_quantile = d3.quantile(data_sorted, .5);
+    let median = d3.median(data_sorted);
+    //window.console.log('median_quantile', median_quantile);
+    //window.console.log('median', median);
+
+    let q3 = d3.quantile(data_sorted, .75);
+    let interQuantileRange = q3 - q1;
+    let min = q1 - (1.5 * interQuantileRange);
+    let max = q3 + (1.5 * interQuantileRange);
+    // min = -1;
+    // median = 0;
+    // max = 1;
+    let minData = Math.min(...data_sorted);
+    let maxData = Math.max(...data_sorted);
+
+    // print stats
+    // window.console.debug('raw key: ', rawKey);
+    // window.console.debug('sorted data: ', data_sorted);
+    // window.console.debug('q1: ', q1);
+    // window.console.debug('q3: ', q3);
+    // window.console.debug('interQuantileRange: ', interQuantileRange);
+    // window.console.debug('median: ', median);
+    // window.console.debug('min (whisker): ', min);
+    // window.console.debug('max (whisker): ', max);
+    // use minimum/max values in data as lower & upper bounds for whisker lines
```

Comment:
For whisker plot, maybe just simply use minData/maxData as lower/upper bounds. `    let min = q1 - (1.5 * interQuantileRange);
    let max = q3 + (1.5 * interQuantileRange);` may be not needed.

Reply:
I made some test plots with plotly.js and have that on a separate branch - https://github.com/EpistasisLab/pennai/tree/plotly_test_for_dataset_page

I posted some screenshots on the slack but just to be thorough I'll upload a few here as well 

![image](https://user-images.githubusercontent.com/6619927/63467215-ec194a00-c432-11e9-87f4-ec1ea3d241d1.png)

The plotly boxplot whisker lines should be configurable; there are three options for specifying a boxplot's boxpoints in plotly - https://plot.ly/javascript/reference/#box-boxpoints